nimbus/src/nimbus.idl

```diff
@@ -87,12 +87,31 @@ interface NimbusClient {
     [Throws=Error]
     sequence<EnrollmentChangeEvent> set_global_user_participation(boolean opt_in);
 
-    // Updates the list of experiments from the server. After calling this, the
-    // list of active experiments might change (there might be new experiments,
-    // or old experiments might have expired)
+    // Updates the list of experiments from the server.
+    // This method is deprecated, in favour of calling `fetch_experiments()` and then
+    // `apply_pending_updates()`.
     [Throws=Error]
     sequence<EnrollmentChangeEvent> update_experiments();
 
+    // Fetches the list of experiments from the server. This does not affect the list
+    // of active experiments or experiment enrolment.
+    // Fetched experiments are not applied until `apply_pending_updates()` is called.
+    [Throws=Error]
+    void fetch_experiments();
+
+    // Apply the updated experiments from the last fetch.
+    // After calling this, the list of active experiments might change
+    // (there might be new experiments, or old experiments might have expired).
+    [Throws=Error]
+    sequence<EnrollmentChangeEvent> apply_pending_updates();
+
+    // A convenience method for apps to set the experiments from a local source
+    // for either testing, or before the first fetch has finished.
+    // 
+    // Experiments set with this method are not applied until `apply_pending_updates()` is called.
+    [Throws=Error]
+    void set_experiments_locally(string experiments_json);
```

Comment:
And here, how about something that's a little clearer about what it's actually doing - `stage_pending_experiments()` springs to mind - it's a littler clearer that this isn't anything new conceptually than what's already described above.

Reply:
For some reason, I don't like `stage` here. We're also using `fetchExperiments()`, and `applyPendingExperiments()`, which would suggest we could use `writeExperiments(string)` or `setPendingExperiments(string)`.

(Notice that we're not accumulating experiments to be applied; one `fetchExperiments` will overwrite another.

If this were a Swift API, I think I'd call this: 

```swift
func setExperiments(from string: String)

// use like: 
nimbusClient.setExperiment(from: """{
    "data": [
         …elided…
     ]
} """)
```

But then perhaps we this method should re-calculate enrollments immediately, without the app having to call `apply` itself.

Reply:
I've left this as is; I like the API called `set_experiments_locally()`, I'm not wild about `set_pending_experiments()` and dislike `stage` anything.